producers/smd_producer.py

```diff
@@ -533,7 +551,7 @@ def get_sd_file(write_dir, exp, hutch):
                     userDict[det._name+'_env']=envData
             except:
                 pass
-            det.processSums()
+            det.processSums(dropped=isDropped(def_data))
```

Comment:
I guess I'm missing something here.  det is a DetObject, no?  So you have one key... dropped, which is either True or False.  But then...

Reply:
Yeah, this will need a write-up: everything about how to use the Sums is a little - well, not obvious.
What I implemented this for is for dropped events so that we can see if the pedestals drifted & a new run should be taken. Now, in the Sums, I'm reading in **kwargs: in this case, the code will find 'dropped' as the key and if the name for the Sums also contains 'dropped', then only events where the value is True will be summed. With this, users could use the same mechanism to e.g. only select events with an I0 above some number and/or laser-on shots,...: as long as they pas s a key with a value of true&false and add a "Sum" where the key is part of the name and they get a sum of all events passing this selection.
Now the two unpretty parts are: there is no simple way (or maybe none...) to save _how_ these events have been selected in the hdf5 file, breaking on of the features that I find pretty important for everything else. The other part is that I'm overlading the name of the "Sum" to contain a _lot_ of information: raw data shape or image, thresholds, simple sum or sum of squares and now even an event selection.